ArduPlane/config.h

```diff
@@ -212,6 +213,14 @@
  #define AP_PLANE_OFFBOARD_GUIDED_SLEW_ENABLED 1
 #endif
 
+#ifndef HAL_QUADPLANE_ENABLED
+#define HAL_QUADPLANE_ENABLED 1
+#endif
+
+#ifndef AP_PLANE_TAILSITTER_ENABLED
+#define AP_PLANE_TAILSITTER_ENABLED (HAL_QUADPLANE_ENABLED && AP_MOTORS_TAILSITTER_ENABLED)
```

Comment:
Tailsitters don't have to use the tailsitter motors backend. "Copter" tailsitters can use quad, hex, octo ect.

I'm not against being able to define out tailsitter, but we can't do it based on the motors backend like this.